data-protocols/dsp/dsp-catalog/dsp-catalog-http-dispatcher/src/main/java/org/eclipse/edc/protocol/dsp/catalog/dispatcher/delegate/DatasetRequestHttpRawDelegate.java

```diff
@@ -42,6 +42,7 @@ public DatasetRequestHttpRawDelegate() {
     public Function<Response, byte[]> parseResponse() {
         return response -> {
             try {
+                assert response.body() != null;
```

Comment:
we do not have asserts anywhere in our code base. maybe better to use a null check and throw an exception?

Reply:
I'd like to avoid doing that because it will pollute the code and these are programming issues that will not happen in correctly structured code.

Reply:
The assertions are messing with the body when they are enabled so I'll replace them with ignore statements to get rid of the warnings.